api/v1beta1/exchange_webhook.go

```diff
@@ -19,8 +19,20 @@ func (r *Exchange) SetupWebhookWithManager(mgr ctrl.Manager) error {
 
 var _ webhook.Validator = &Exchange{}
 
-// no validation on create
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+// either rabbitmqClusterReference.name or rabbitmqClusterReference.connectionSecret must be provided but not both
 func (e *Exchange) ValidateCreate() error {
+	if e.Spec.RabbitmqClusterReference.Name != "" && e.Spec.RabbitmqClusterReference.ConnectionSecret != nil {
+		return apierrors.NewForbidden(e.GroupResource(), e.Name,
+			field.Forbidden(field.NewPath("spec", "rabbitmqClusterReference"),
+				"do not provide both spec.rabbitmqClusterReference.name and spec.rabbitmqClusterReference.connectionSecret"))
+	}
+
+	if e.Spec.RabbitmqClusterReference.Name == "" && e.Spec.RabbitmqClusterReference.ConnectionSecret == nil {
+		return apierrors.NewForbidden(e.GroupResource(), e.Name,
+			field.Forbidden(field.NewPath("spec", "rabbitmqClusterReference"),
+				"must provide either spec.rabbitmqClusterReference.name or spec.rabbitmqClusterReference.connectionSecret"))
+	}
 	return nil
```

Comment:
Could we extract this test case into a common function used by all of the webhooks?